provider/pkg/custom/interface.go

```diff
@@ -0,0 +1,18 @@
+package custom
```

Comment:
I wanted the ExtensionResource here, but due to the structure of our other existing modules, it would create import cycles, so I had to move it to `schema` for now. But moving the interface into schema felt more wrong so I left it here in the hope we can fix our modules in follow-up work and move it back.